lib/trino/client/statement_client.rb

```diff
@@ -66,20 +66,46 @@ def init_request(req)
 
     def post_query_request!
       uri = "/v1/statement"
-      response = @faraday.post do |req|
-        req.url uri
 
-        req.body = @query
-        init_request(req)
-      end
+      start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      attempts = 0
 
-      # TODO error handling
-      if response.status != 200
-        exception! TrinoHttpError.new(response.status, "Failed to start query: #{response.body} (#{response.status})")
+      loop do
+        begin
+          response = @faraday.post do |req|
+            req.url uri
+            req.body = @query
+            init_request(req)
+          end
+        rescue Faraday::TimeoutError, Faraday::ConnectionFailed
+          # temporally error to retry
+          response = nil
+        rescue => e
+          exception! e
+        end
+
+        if response
+          if response.status == 200 && !response.body.to_s.empty?
```

Comment:
```suggestion
          if response.status == 200
```

Please delete `&& !response.body.to_s.empty?`. It may be required, but did not exist in the previous code.
We may add it when it found needed.

Reply:
Hi, exoego. I removed `&& !response.body.to_s.empty?`.

Reply:
Thanks